### DISCUSSION TITLE
Custom CSS for Servers

### DISCUSSION THREAD

**Discussion by Pseudoexpertise** [stoatchat/discussions#525] [Feature Suggestions]
## Suggestion
Whatever you do will only be visible on your device. But what if you want the server's theme to fit to your topic?
Other than that I was suggesting a certain functionality in my previous post (#524). I was suggesting to 
optionally minimize the left sidebar because you don't need the channel's names if you can distinct them by their icons.
Well, _if and only if_! Otherwise it won't make sense at all! This feature does only make sense for those who are active in servers that are designed by an approach similar to mine.

## Example
Now imagine joining a retro gaming community and there is Zelda in the background, a little Mario in an animated transition pushing the channel name in the header (class="Header-sc-ujh2d9-0 gXyCoD") to the right place ... wouldn't that be magical?
The only limitations are your creativity (and the servers capacity).

## Problems
In contrast, forcing other members to adapt to the styling some owner made is not a good idea at all! There are those who are bad at designing and those who are brilliant but have a terrible taste!
We need an option to enable or disable Server-styling.

However there is one thing I am unshure about:
If you have custom CSS but are enabling, call it, *server CSS*, who is overwriting who?
Should we be able to decide in our settings whether the server CSS, if enabled, overwrites ours or if ours will overwrite the server's when dealing with CSS conflicts?

**Comment by ghost** [comment]
Not a bad idea actually, that's allow community to use the best of revolt

**Comment by iGerman00** [comment]
Definitely needs to be client-side controllable and with reasonable limits as to what is stylable and in what why. A toggle for displaying the styles is a requirement for sure.

**Comment by nulldg** [comment]
servers should be free to stylize themselves, but i don't think CSS is the right way to go. a naive solution may lead to visual conflicts. for instance, if a server overwrote some text's color to be white but left the background color alone, then it would be invisible on white themes.

a lot of thought would have to go into this problem.
